test/e2e/daemon_set.go

```diff
@@ -48,7 +48,14 @@ const (
 )
 
 var _ = Describe("Daemon set", func() {
-	f := &Framework{BaseName: "daemonsets"}
+	var f *Framework
+
+	AfterEach(func() {
```

Comment:
I'd prefer not to invent another ordering behavior on top of what Ginkgo already provides.  However, I agree the ordering is non obvious in some cases.

Framework invoking before/after directly is part of the issue - I wonder if framework before each can set an after test for itself (so that the framework after is guaranteed to be called last).